src/components/SchedulePanel/components/OfficeHoursDialog/index.js

```diff
@@ -0,0 +1,526 @@
+      officeHoursOpen: false,
+      checkedC: false,
+      checkedDayofWeek: {
+        checkedMo: false,
+        checkedTu: false,
+        checkedWe: false,
+        checkedTh: false,
+        checkedFr: false,
+        checkedSa: false,
+        checkedSu: false,
+      },
+      selectedEvent: null,
+      descriptionInput: '',
+      locationInput: '',
+      descriptionValid: false,
+      locationValid: false,
+      startHourValid: true,
+      endHourValid: true,
+      formValid: false,
+      checkedValid: false,
+      formErrors: {
+        descriptionInput: '',
+        locationInput: '',
+        startHourValid: '',
+        endHourValid: '',
+      },
+    };
+    this.handleOfficeHoursClose = this.handleOfficeHoursClose.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleReset = this.handleReset.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    // Case : Edit(doubleclick)
+    if (this.state.selectedEvent !== nextProps.selectedEvent && nextProps.selectedEvent) {
+      this.setState({ selectedEvent: nextProps.selectedEvent }, () => {
+        myschedule
+          .getMyScheduleEventId(this.state.selectedEvent.id)
+          .then(event => {
+            this.checkDay(event);
+            this.checkTime(event);
+            let eventDescLoc = this.state.selectedEvent.title.split(' in ');
+            this.setState({
+              descriptionInput: eventDescLoc[0],
+              locationInput: eventDescLoc[1],
+              descriptionValid: true,
+              locationValid: true,
+              startHourValid: true,
+              endHourValid: true,
+              formValid: true,
+              checkedValid: true,
+            });
+          })
+          .catch(error => {
+            console.log(error);
+          });
+      });
+      // Case : Add(selecslot)
+    } else {
+      this.handleReset();
+      this.setState({
+        checkedDayofWeek: {
+          checkedSu: nextProps.resourceId === 1 ? true : false,
+          checkedMo: nextProps.resourceId === 2 ? true : false,
+          checkedTu: nextProps.resourceId === 3 ? true : false,
+          checkedWe: nextProps.resourceId === 4 ? true : false,
+          checkedTh: nextProps.resourceId === 5 ? true : false,
+          checkedFr: nextProps.resourceId === 6 ? true : false,
+          checkedSa: nextProps.resourceId === 7 ? true : false,
+        },
+        startHourInput:nextProps.startTime,
+        endHourInput:nextProps.endTime,
+        checkedValid: true,
+        startHourValid: true,
+        endHourValid: true,
+      });
+    }
+  }
+
+  validateField(fieldName, value) {
+    let fieldValidationErrors = this.state.formErrors;
+    let descriptionValid = this.state.descriptionValid;
+    let locationValid = this.state.locationValid;
+    let startHourValid = this.state.startHourValid;
+    let endHourValid = this.state.endHourValid;
+
+    // Require that content havint appropriate size and format
```

Comment:
What does "havint" mean here? Is it supposed to be "have"?

Reply:
Change made